iceoryx_posh/source/roudi/roudi_cmd_line_parser.cpp

```diff
@@ -24,20 +24,20 @@ namespace iox
 {
 namespace config
 {
-void CmdLineParser::parse(int argc, char* argv[], const CmdLineArgumentParsingMode cmdLineParsingMode) noexcept
+cxx::expected<CmdLineArgs_t, CmdLineParserResult>
+CmdLineParser::parse(int argc, char* argv[], const CmdLineArgumentParsingMode cmdLineParsingMode) noexcept
 {
     constexpr option longOptions[] = {{"help", no_argument, nullptr, 'h'},
                                       {"version", no_argument, nullptr, 'v'},
                                       {"monitoring-mode", required_argument, nullptr, 'm'},
                                       {"log-level", required_argument, nullptr, 'l'},
-                                      {"ignore-version", required_argument, nullptr, 'i'},
                                       {"unique-roudi-id", required_argument, nullptr, 'u'},
-                                      {"compatibility", required_argument, nullptr, 'c'},
+                                      {"compatibility", required_argument, nullptr, 'x'},
                                       {"kill-delay", required_argument, nullptr, 'k'},
                                       {nullptr, 0, nullptr, 0}};
 
     // colon after shortOption means it requires an argument, two colons mean optional argument
-    constexpr const char* shortOptions = "hvm:l:u:c:k:";
+    constexpr const char* shortOptions = "hvm:l:u:x:k:";
```

Comment:
Hope this change won't lead to misfortune :laughing:

Reply:
oh no, the luck is gone :laughing: